### ISSUE TITLE
SubscriptionWsProtocol toMessageMap cast error

### ISSUE THREAD

**Issue by RamiroEda** [apollographql/apollo-kotlin#4061]
**Summary**
When I call "collect" I get a network error due to a casting error in `WsProtocol.toMessageMap`.

**Version**
3.2.2

**Description**
I think the Date is causing the error but I don't know if the client should handle that value. 
Even so, I publish it in case someone finds this issue useful.
![image](https://user-images.githubusercontent.com/20670965/164959954-40eec5dc-5709-4ed9-bcdd-884658d7ab9a.png)
![image](https://user-images.githubusercontent.com/20670965/164959963-ae11ee21-8e57-41ee-a1cd-29a92828830a.png)

This is my client builder
```kt
ApolloClient.Builder()
        .serverUrl(ENDPOINT_URL)
        .webSocketServerUrl(PUBSUB_ENDPOINT_URL)
        .normalizedCache(
            SqlNormalizedCacheFactory(
                app,
                "apollo_db"
            ).chain(SqlNormalizedCacheFactory(app, "apollo_cache"))
        )
        .httpEngine(DefaultHttpEngine(httpClient))
        .wsProtocol(SubscriptionWsProtocol.Factory(
            connectionPayload = {
                mapOf(
                    "token" to "Bearer ${Preferences(app).getPreference(PreferenceKeys.AccessToken)}"
                )
            }
        ))
        .build()
```
Stacktrace
```plaintext
W/System.err: com.apollographql.apollo3.exception.ApolloNetworkException: Network error while executing OnChatsUnreadCountChanged
W/System.err:     at com.apollographql.apollo3.network.ws.WebSocketNetworkTransport$execute$$inlined$map$1$2.emit(Collect.kt:144)
W/System.err:     at kotlinx.coroutines.flow.internal.SafeCollectorKt$emitFun$1.invoke(SafeCollector.kt:15)
W/System.err:     at kotlinx.coroutines.flow.internal.SafeCollectorKt$emitFun$1.invoke(SafeCollector.kt:15)
W/System.err:     at kotlinx.coroutines.flow.internal.SafeCollector.emit(SafeCollector.kt:77)
W/System.err:     at kotlinx.coroutines.flow.internal.SafeCollector.emit(SafeCollector.kt:59)
W/System.err:     at com.apollographql.apollo3.network.ws.WebSocketNetworkTransport$execute$3.invokeSuspend(WebSocketNetworkTransport.kt:252)
W/System.err:     at com.apollographql.apollo3.network.ws.WebSocketNetworkTransport$execute$3.invoke(Unknown Source:13)
W/System.err:     at com.apollographql.apollo3.network.ws.WebSocketNetworkTransport$execute$3.invoke(Unknown Source:6)
W/System.err:     at com.apollographql.apollo3.internal.FlowsKt$transformWhile$1$invokeSuspend$$inlined$collectWhile$1.emit(flows.kt:53)
W/System.err:     at com.apollographql.apollo3.internal.FlowsKt$transformWhile$1$invokeSuspend$$inlined$collectWhile$2.emit(Collect.kt:135)
W/System.err:     at com.apollographql.apollo3.network.ws.WebSocketNetworkTransport$execute$$inlined$filter$1$2.emit(Collect.kt:137)
W/System.err:     at kotlinx.coroutines.flow.SubscribedFlowCollector.emit(Unknown Source:2)
W/System.err:     at kotlinx.coroutines.flow.SharedFlowImpl.collect$suspendImpl(SharedFlow.kt:372)
W/System.err:     at kotlinx.coroutines.flow.SharedFlowImpl$collect$1.invokeSuspend(Unknown Source:15)
W/System.err:     at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
W/System.err:     at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
W/System.err:     at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:39)
W/System.err:     at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
W/System.err:     at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
W/System.err:     at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
W/System.err:     at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
W/System.err:     at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
W/System.err: Caused by: java.lang.ClassCastException: java.lang.String cannot be cast to java.util.Map
W/System.err:     at com.apollographql.apollo3.network.ws.WsProtocol.toMessageMap(WsProtocol.kt:96)
W/System.err:     at com.apollographql.apollo3.network.ws.WsProtocol.receiveMessageMap(WsProtocol.kt:116)
W/System.err:     at com.apollographql.apollo3.network.ws.WsProtocol$receiveMessageMap$1.invokeSuspend(Unknown Source:14)
W/System.err:     at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
W/System.err:     at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
W/System.err:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
W/System.err:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
W/System.err:     at java.lang.Thread.run(Thread.java:920)
```

**Comment by martinbonnin** [comment]
This is unexpected indeed. Can you share the details of your backend? Is it publicly available by any chance? Feel free to send details at martin@apollographql.com or here if you prefer.

**Comment by RamiroEda** [comment]
I dont have access to the backend code but is a Node JS server (Apollo Server 3) with Schema Stitching to a Laravel Server (Lighthouse).
The Node server handle all subscriptions and the Laravel server handles Queries and Mutations.
This is the URL to the dev server we have https://gmar-node.herokuapp.com/graphql.
Is worth to mention that this was not an issue in V2.

**Comment by martinbonnin** [comment]
Thanks for the details and sorry for the delay! Looks like V2 just ignored these messages. I'll push a fix for this.

**Comment by RamiroEda** [comment]
Thanks for the help!

**Comment by martinbonnin** [comment]
Any chance you can try `3.2.2-SNAPSHOT` (https://github.com/apollographql/apollo-kotlin#snapshots) ? This way we can confirm it's working before next release?

**Comment by RamiroEda** [comment]
It works as intended, thank you

**Comment by martinbonnin** [comment]
Thanks for the update!

**Comment by BoD** [comment]
Available in v3.3.0.
